### PR TITLE
feat!: Remove Python 2 support

### PR DESCRIPTION
**Pull request by andrewsg**
BREAKING CHANGE: This PR removes Python 2 support in the setup.py metadata, test runner and code itself. This is intended to be released as a major version update. Python 2 users can continue to use the 1.x version of the client library.

### CODE REVIEW

File: google/cloud/storage/_signing.py [hunk 1]
```diff
@@ -20,7 +20,8 @@
 import hashlib
 import json
 
-import six
+import http
```
Comment (tseaver) [inline google/cloud/storage/_signing.py:23]: Prefer to import the explicit submodule:
```suggestion
import http.client
```

Also, sort w/ stdlib imports.

File: google/cloud/storage/batch.py [hunk 2]
```diff
@@ -65,13 +64,8 @@ def __init__(self, method, uri, headers, body):
...
         payload = "\r\n".join(lines)
-        if six.PY2:
-            # email.message.Message is an old-style class, so we
-            # cannot use 'super()'.
-            MIMEApplication.__init__(self, payload, "http", encode_noop)
-        else:  # pragma: NO COVER  Python3
-            super_init = super(MIMEApplicationHTTP, self).__init__
-            super_init(payload, "http", encode_noop)
+        super_init = super(MIMEApplicationHTTP, self).__init__
+        super_init(payload, "http", encode_noop)
```
Comment (tseaver) [inline google/cloud/storage/batch.py:68]: For Python3-only, `super()` is nice:
```suggestion
        super()__init__(payload, "http", encode_noop)
```

File: google/cloud/storage/_helpers.py [hunk 3]
```diff
@@ -19,11 +19,9 @@
 
 import base64
 from hashlib import md5
-from datetime import datetime
 import os
 
-from six import string_types
-from six.moves.urllib.parse import urlsplit
+from urllib.parse import urlsplit
```
Comment (tseaver) [inline google/cloud/storage/_helpers.py:24]: Sort w/ stdlib imports.

File: google/cloud/storage/blob.py [hunk 4]
```diff
@@ -37,12 +37,11 @@
...
-from six.moves.urllib.parse import parse_qsl
-from six.moves.urllib.parse import quote
-from six.moves.urllib.parse import urlencode
-from six.moves.urllib.parse import urlsplit
-from six.moves.urllib.parse import urlunsplit
+from urllib.parse import parse_qsl
+from urllib.parse import quote
+from urllib.parse import urlencode
+from urllib.parse import urlsplit
+from urllib.parse import urlunsplit
```
Comment (tseaver) [inline google/cloud/storage/blob.py:44]: Sort w/ stdlib imports.

File: google/cloud/storage/bucket.py [hunk 5]
```diff
@@ -20,8 +20,7 @@
 import json
 import warnings
 
-import six
-from six.moves.urllib.parse import urlsplit
+from urllib.parse import urlsplit
```
Comment (tseaver) [inline google/cloud/storage/bucket.py:23]: Sort w/ stdlib imports.

File: tests/unit/test__signing.py [hunk 6]
```diff
@@ -24,8 +24,7 @@
 
 import mock
 import pytest
-import six
-from six.moves import urllib_parse
+import urllib.parse
```
Comment (tseaver) [inline tests/unit/test__signing.py:27]: Sort w/ stdlib imports.

File: tests/unit/test_batch.py [hunk 7]
```diff
@@ -16,7 +16,10 @@
 
 import mock
 import requests
-from six.moves import http_client
+import http.client
+
+from http.client import SERVICE_UNAVAILABLE
+from http.client import NO_CONTENT
```
Comment (tseaver) [inline tests/unit/test_batch.py:22]: Sort w/ stdlib imports.

File: tests/unit/test_blob.py [hunk 8]
```diff
@@ -20,12 +20,11 @@
 import os
 import tempfile
 import unittest
+import http.client
```
Comment (tseaver) [inline tests/unit/test_blob.py:23]: Sort under `hashlib`

File: tests/unit/test_client.py [hunk 9]
```diff
@@ -20,9 +20,9 @@
 import re
 import requests
 import unittest
-from six import string_types
-from six.moves import http_client
-from six.moves.urllib import parse as urlparse
+import http.client
```
Comment (tseaver) [inline tests/unit/test_client.py:23]: Sort below `base64`

**Review by tseaver — CHANGES_REQUESTED** [review]

**Comment by tseaver** [comment]
@andrewsg The `Kokoro system-2.7` job is configured both in the `.kokoro/presubmit/system-2.7.cfg` file and in `google3`.

**Comment by andrewsg** [comment]
@tseaver This was on hold due to the release freeze last month. Thank you as always for your thorough review. PTAL. Will address google3 issue shortly.

File: README.rst [hunk 10]
```diff
@@ -65,6 +65,9 @@ Unsupported Python Versions
 Python == 3.5: the last released version which supported Python 3.5 was
 ``google-cloud-storage 1.32.0``, released 2020-10-16.
 
+Python == 2.7: the last released version which supported Python 2.7 was
+``google-cloud-storage 1.43.0``, released 2021-11-17.
```
Comment (cojenco) [inline README.rst:69]:
```suggestion
``google-cloud-storage 1.44.0``, released 2022-01-05.
```

**Comment by Breathtender** [comment]
So happy to see all the clean up. Nothing else to add beyond the requests that tres and cathy made; lgtm. Is there a change that should be made in here for kokoro to not try and still run the system-2.7 tests?

**Review by cojenco — COMMENTED** [review]
LGTM🎉🎉 Thanks Andrew for getting us across the finish line!

**Review by cojenco — APPROVED** [review]

**Review by tseaver — APPROVED** [review]

**Pull request merged**